### PR TITLE
Add Theme branch into publish manifest

### PR DESCRIPTION
**Pull request by 928PJY**
[AB#319985](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/319985)

### CODE REVIEW

File: docs/specs/template.yml [hunk 1]
```diff
@@ -15,6 +15,22 @@ repos:
 outputs:
   docs/a.json: |
     { "conceptual": "<div class=\"NOTE\"><p>my note</p><p>note</p></div>" }
+  .publish.json: |
+    {"theme_branch": "main"}
+---
```
Comment (herohua) [inline docs/specs/template.yml:20]: shall we add a case of none-master/main branch?

Comment (928PJY) [reply docs/specs/template.yml:20]: updated

File: src/docfx/template/TemplateEngine.cs [hunk 2]
```diff
@@ -63,6 +66,19 @@ public RenderType GetRenderType(ContentType contentType, SourceInfo<string?> mim
             };
         }
 
+        public string? GetThemeBranch()
+        {
+            if (_package is LocalPackage package)
+            {
+                var (themeRepo, _) = _repositoryProvider.GetRepository(PathString.DangerousCreate($"{package.PackagePath}/"));
```
Comment (herohua) [inline src/docfx/template/TemplateEngine.cs:73]: > _repositoryProvider.GetRepository(PathString.DangerousCreate($"{package.PackagePath}/")); [](start = 36, length = 90)

This looks like internal logic of `LocalPackage`. Would it be better let package returns branch info?

**Review by partychen — APPROVED** [review]

File: src/docfx/template/TemplateEngine.cs (hunk 2 above)
Comment (928PJY) [reply src/docfx/template/TemplateEngine.cs:73]: but the package is not always a repository, even for localPackage, it is more related to the template, so I put it here.

**Review by herohua — APPROVED** [review]
:shipit:

File: src/docfx/publish/PublishModelBuilder.cs [hunk 3]
```diff
@@ -79,6 +82,7 @@ orderby monikerGroup
                 _config.Name,
                 _config.Product,
                 _config.BasePath.ValueWithLeadingSlash,
+                _templateEngine.GetThemeBranch(),
```
Comment (yufeih) [inline src/docfx/publish/PublishModelBuilder.cs:85]: Is it suffice enough to say:
```suggestion
                _config.Template.Branch,
```

**Comment by yufeih** [comment]
> No pipelines are associated with this pull request.

Hit the same problem described here: https://developercommunity.visualstudio.com/content/problem/1211721/azp-run-does-not-find-associated-pipelines-for-pr.html

**Review by yufeih — APPROVED** [review]

**Pull request merged**